src/lib/assets/assetsDeployer.ts

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023-Present The Pepr Authors
+
+import { AssetsConfig } from "./assetsConfig";
+import { deploy } from "./deploy";
+import { loadCapabilities } from "./loader";
+import { allYaml, zarfYaml, zarfYamlChart, overridesFile } from "./yaml";
+import { dumpYaml } from "@kubernetes/client-node";
+import { createDirectoryIfNotExists, dedent, namespaceComplianceValidator, replaceString } from "../helpers";
+import { resolve } from "path";
+import {
+  chartYaml,
+  nsTemplate,
+  admissionDeployTemplate,
+  watcherDeployTemplate,
+  clusterRoleTemplate,
+  serviceMonitorTemplate,
+} from "./helm";
+import { promises as fs } from "fs";
+import { webhookConfig } from "./webhooks";
+import { apiTokenSecret, service, tlsSecret, watcherService } from "./networking";
+import { watcher, moduleSecret } from "./pods";
+
+import { clusterRoleBinding, serviceAccount, storeRole, storeRoleBinding } from "./rbac";
+
+export class AssetsDeployer {
+  constructor(private assetsConfig: AssetsConfig) {}
+
+  async deploy(force: boolean, webhookTimeout?: number) {
+    // Load capabilities and trigger the deploy logic
+    this.assetsConfig.capabilities = await loadCapabilities(this.assetsConfig.path);
+    await deploy(this.assetsConfig, force, webhookTimeout);
+  }
+
+  async allYaml(imagePullSecret?: string) {
+    this.assetsConfig.capabilities = await loadCapabilities(this.assetsConfig.path);
+
+    // Namespace validation and generating YAMLs
+    for (const capability of this.assetsConfig.capabilities) {
+      namespaceComplianceValidator(capability, this.assetsConfig.alwaysIgnore?.namespaces);
+    }
+
+    return allYaml(this.assetsConfig, imagePullSecret);
+  }
+
+  async zarfYaml(path: string) {
+    return zarfYaml(this.assetsConfig, path);
+  }
+
+  async zarfYamlChart(path: string) {
```

Comment:
doesn't this just return some json? not async needed unless i am missing something

Reply:
Fixed.